runtime/parallel/src/lib.rs

```diff
@@ -1195,11 +1203,115 @@ parameter_types! {
     pub DefaultProtocolFeeReceiver: AccountId = TreasuryPalletId::get().into_account();
 }
 
+pub struct Adapter<AccountId> {
```

Comment:
Could we add a supporting pallet like orml-currencies to handle the duplicate code?

Reply:
Sorry @yz89 I didn't understand this comment. Could you please explain more?

cc @ETeissonniere

Reply:
We defined the `pub struct Adapter<AccountId>` in heiko, parallel and vanilla runtime. I mean if we can extract the logic into a pallet.